core/src/test/java/org/jobrunr/scheduling/interval/IntervalTest.java

```diff
@@ -0,0 +1,119 @@
+package org.jobrunr.scheduling.interval;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.stream.Stream;
+
+import static java.time.ZoneOffset.UTC;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+class IntervalTest {
+
+    private static final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+
+    private static final String TEN_SECONDS = "PT10S";
+    private static final String FORTY_EIGHT_HOURS = "PT48H";
+    private static final String EIGHT_DAYS = "P8D";
+
+    @ParameterizedTest
+    @MethodSource("startInstantDurationAndResultInstant")
+    void testInterval(String baseDate, String durationExpression) {
+        try {
+            Instant inputInstant = LocalDateTime.parse(baseDate, dateTimeFormatter).toInstant(UTC);
+            Interval interval = new Interval(durationExpression);
+            Duration duration = Duration.parse(durationExpression);
+            Instant nextInstant = interval.next(inputInstant, UTC);
+            Instant now = Instant.now();
+
+            assertThat(nextInstant)
+                    .describedAs("Expecting %s to be after or equal to %s for duration %s and start date %s", nextInstant, now, duration, inputInstant)
+                    .isAfterOrEqualTo(now);
+            assertThat(nextInstant)
+                    .describedAs("Expecting %s to be before to %s for duration %s and start date %s", nextInstant, now.plus(duration), duration, inputInstant)
+                    .isBefore(now.plus(duration));
+        } catch (Exception e) {
+            System.out.printf("Error for %s and %s%n", baseDate, durationExpression);
+            throw e;
+        }
+    }
```

Comment:
This test can now accept to more arguments:
`arguments("2019-01-01 00:00:00", TEN_SECONDS, "2019-01-5 00:00:01", "2019-01-5 00:00:10"),`

This would thus be:
"2019-01-01 00:00:00" => baseDate
TEN_SECONDS => intervalExpression
"2019-01-5 00:00:00" => the moment when JobRunr is evaluating the expression (Instant.now() in production code)
"2019-01-5 00:00:10" => the expected outcome

Now, we can also really test corner cases to the exact moment.